staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -105,9 +106,26 @@ func NewGenericAdmissionWebhook(configFile io.Reader) (*GenericAdmissionWebhook,
 		),
 		authInfoResolver: authInfoResolver,
 		serviceResolver:  defaultServiceResolver{},
+		cache:            &clientCache{clientMap: make(map[admissionHookClientConfig]*rest.RESTClient)},
 	}, nil
 }
 
+type admissionHookClientConfig struct {
+	Service v1alpha1.ServiceReference
+
+	// URLPath is an optional field that specifies the URL path to use when posting the AdmissionReview object.
+	URLPath string
+
+	// CABundle is a PEM encoded CA bundle which will be used to validate webhook's server certificate.
+	// Required
+	CABundle string
+}
+
+type clientCache struct {
+	clientMap map[admissionHookClientConfig]*rest.RESTClient
```

Comment:
should be an lrucache with a max size to avoid growing without bound

Reply:
ok